crates/evm/fuzz/src/strategies/mutators.rs

```diff
@@ -0,0 +1,482 @@
+use alloy_primitives::{Address, I256, Sign, U256};
+use proptest::{prelude::*, test_runner::TestRunner};
+use rand::seq::IndexedRandom;
+
+// Interesting 8-bit values to inject.
+static INTERESTING_8: [i8; 9] = [-128, -1, 0, 1, 16, 32, 64, 100, 127];
+
+/// Interesting 16-bit values to inject.
+static INTERESTING_16: [i16; 19] = [
+    -128, -1, 0, 1, 16, 32, 64, 100, 127, -32768, -129, 128, 255, 256, 512, 1000, 1024, 4096, 32767,
+];
+
+/// Interesting 32-bit values to inject.
+static INTERESTING_32: [i32; 27] = [
+    -128,
+    -1,
+    0,
+    1,
+    16,
+    32,
+    64,
+    100,
+    127,
+    -32768,
+    -129,
+    128,
+    255,
+    256,
+    512,
+    1000,
+    1024,
+    4096,
+    32767,
+    -2147483648,
+    -100663046,
+    -32769,
+    32768,
+    65535,
+    65536,
+    100663045,
+    2147483647,
+];
+
+/// Mutator that randomly increments or decrements an uint or int.
+pub(crate) trait IncrementDecrementMutator: Sized + Copy {
+    fn increment_decrement(self, size: usize, rng: &mut TestRunner) -> Option<Self>;
+}
+
+macro_rules! impl_increment_decrement {
+    ($($t:ty => $validate_fn:path),*) => {
+        $(
+            impl IncrementDecrementMutator for $t {
+                fn increment_decrement(self, size: usize, test_runner: &mut TestRunner) -> Option<Self> {
+                    let mutated = if test_runner.rng().random::<bool>() {
+                        self.wrapping_add(Self::ONE)
+                    } else {
+                        self.wrapping_sub(Self::ONE)
+                    };
+                    Self::validate(self, mutated, size)
+                }
+            }
+        )*
+    };
+}
+
+impl_increment_decrement!(
+    U256 => validate_uint_mutation,
+    I256 => validate_int_mutation
+);
+
+/// Mutator that flips random bit of uint, int or address.
+pub(crate) trait BitFlipMutator: Sized + Copy {
+    fn flip_random_bit(
+        value: Self,
+        size: Option<usize>,
+        test_runner: &mut TestRunner,
+    ) -> Option<Self>;
+}
+
+impl BitFlipMutator for U256 {
+    fn flip_random_bit(
+        value: Self,
+        size: Option<usize>,
+        test_runner: &mut TestRunner,
+    ) -> Option<Self> {
+        let bit_index = test_runner.rng().random_range(0..size?);
+        let mask = Self::from(1u8) << bit_index;
+        let flipped = value ^ mask;
+        validate_uint_mutation(value, flipped, size?)
+    }
+}
+
+impl BitFlipMutator for I256 {
+    fn flip_random_bit(value: Self, size: Option<usize>, rng: &mut TestRunner) -> Option<Self> {
+        let bit_index = rng.rng().random_range(0..size?);
+        let (sign, mut abs): (Sign, U256) = value.into_sign_and_abs();
+        abs ^= U256::from(1u8) << bit_index;
+        let flipped = Self::checked_from_sign_and_abs(sign, abs)?;
+        validate_int_mutation(value, flipped, size?)
+    }
+}
+
+impl BitFlipMutator for Address {
```

Comment:
I think all impls here could be simplified to operate on `&mut [u8]`.

- with a generic low level interface/function of `(&mut [u8], rng) -> Option<()>`
- and a high level one of `(self, size, rng) -> Option<Self>` which can also validate; this can be instrumented with tracing `#[instrument(ret)]` which will log the mutation on return

U256/I256 will pass `&mut be_bytes[..size?]` to the low level mutator thats shared across all types

does this make sense?

Reply:
updates in https://github.com/foundry-rs/foundry/pull/11061/commits/cdbd275cf37a64a53c8ccbf1bf04319b84d6fe1e